pkg/crypto/certloader/server.go

```diff
@@ -134,3 +137,13 @@ func (c *WatchedServerConfig) ServerConfig(base *tls.Config) *tls.Config {
 		MinVersion: tls.VersionTLS13,
 	}
 }
+
+// appendH2Proto appends h2 to the list of protocols if it's not already there.
+func (c *WatchedServerConfig) appendH2Proto(existingProtocols []string) []string {
+	for _, p := range existingProtocols {
+		if p == alpnProtocolH2 {
+			return existingProtocols
+		}
+	}
+	return append(existingProtocols, alpnProtocolH2)
```

Comment:
Could we please avoid modifying the caller provided slice?

Reply:
Sure, updated.